sentry/src/main/java/io/sentry/SentryClientFactory.java

```diff
@@ -100,7 +118,12 @@ public static SentryClient sentryClient(Dsn dsn, String sentryClientFactoryName)
             logger.debug("Attempting to use '{}' as a SentryClientFactory.", sentryClientFactory);
             triedFactories.add(name);
             try {
-                SentryClient sentryClientInstance = sentryClientFactory.createSentryClient(dsn);
+                // ensure each iteration checks its own lookupDsn method if needed
+                Dsn factoryDsn = dsn;
+                if (factoryDsn == null) {
+                    factoryDsn = sentryClientFactory.lookupDsn();
+                }
```

Comment:
This is where the lookups got moved -- so that factories have the power to do their own logic.